frontend/src/components/common/multiselect-input/multiselect-input.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { FC } from 'react';
+
+import { AutocompleteValueType } from '@common/types/cars/autocomplete.type';
+import { Autocomplete, SvgIcon, TextField } from '@mui/material';
+
+import styles from './styles.module.scss';
+
+interface Props {
+  label: string;
+  filterName?: string;
+  options: AutocompleteValueType[];
+  value: AutocompleteValueType[];
+  onChange: (data: { filterName: string; list: string[] }) => void;
+}
+
+const MultiselectInput: FC<Props> = ({
+  label,
+  filterName,
+  options,
+  value,
+  onChange,
+}): JSX.Element => {
+  const handleSelectChange = (values: AutocompleteValueType[]): void => {
+    const ids = values.map((value) => value?.id || '');
+
+    onChange({ filterName: filterName || '', list: ids });
+  };
+
+  const uniqueOptions = options.filter(
+    (option) => !value.map((item) => item?.id).includes(option?.id),
+  );
```

Comment:
Why don't use memoize the result? It will be reevaluated on each rerender.
Also you could use the lodash uniq function here